cfme/scripting/setup_ansible.py

```diff
@@ -0,0 +1,118 @@
+enabled_metadata=1
+EOF"""
+PGDG_94_CENTOS_REPO_PATH = "/etc/yum.repos.d/pgdg-94-centos.repo"
+PGDG_94_CENTOS_REPO_CONTENT = """EOF
+[pgdg94]
+name=PostgreSQL 9.4 $releasever - $basearch
+baseurl=http://download.postgresql.org/pub/repos/yum/9.4/redhat/rhel-\$releasever-\$basearch
+enabled=1
+gpgcheck=0
+EOF"""
+REPOS = [
+    (ANSIBLE_TOWER_REPO_PATH, ANSIBLE_TOWER_REPO_CONTENT),
+    (JLASKA_RABBITMQ_REPO_PATH, JLASKA_RABBITMQ_REPO_CONTENT),
+    (PGDG_94_CENTOS_REPO_PATH, PGDG_94_CENTOS_REPO_CONTENT)
+]
+
+
+def run_command(app, command):
+    print(command)
+    app.ssh_client.run_command(command)
+
+
+def setup_repos(app):
+    print("Starting configuring repos")
+    for path, content in REPOS:
+        run_command(app, "cat > {path} << {content}".format(path=path, content=content))
+
+
+def install_packages(app):
+    print("Starting to install packages")
+    run_command(app, "yum -y install ansible-tower-setup ansible-tower-server")
+    run_command(app, "install -o awx -g awx -m 0755 -d /var/log/tower")
+    run_command(app, "install -o awx -g awx -m 0644 /dev/null /var/log/tower/callback_receiver.log")
+    run_command(app, "install -o awx -g awx -m 0644 /dev/null /var/log/tower/fact_receiver.log")
+    run_command(app, "install -o awx -g awx -m 0644 /dev/null /var/log/tower/task_system.log")
+    run_command(app, "install -o awx -g awx -m 0644 /dev/null /var/log/tower/tower.log")
+    run_command(app,
+        "install -o awx -g awx -m 0644 /dev/null /var/log/tower/tower_rbac_migrations.log")
+    run_command(app,
+        "install -o awx -g awx -m 0644 /dev/null /var/log/tower/tower_system_track_migrations.log")
+    run_command(app,
+        "install -o awx -g awx -m 0644 /dev/null "
+        "/var/log/tower/tower_system_tracking_migrations.log")
+
+
+def get_ansible_password(app):
+    print("Getting embedded ansible password. It can take a long time.")
+    app.ssh_client.run_rails_command("'EmbeddedAnsible.start'")
+    return app.ssh_client.run_rails_command(
+        "'puts MiqDatabase.first.ansible_admin_authentication.password'").output.rstrip()
+
+
+def open_port(app):
+    run_command(app, "firewall-cmd --zone manageiq --add-port {}/tcp".format(ports.TOWER))
+    run_command(app, "firewall-cmd --reload")
+    print("Waiting until port {} will be opened".format(ports.TOWER))
+    wait_for(
+        net_check,
+        [ports.TOWER, app.address],
+        {"force": True},
+        num_sec=600,
+        delay=5
+    )
+
+
+def upload_license(app, license_path):
+    password = get_ansible_password(app)
+    print("Password is {}".format(password))
+    open_port(app)
+    print("Sending license file to embedded ansible")
+    r = requests.post(
+        'https://{addr}:{port}/api/v1/config/'.format(addr=app.address, port=ports.TOWER),
+        headers={"Content-Type": "application/json"},
+        data=open(license_path, "rb"),
+        auth=("admin", password),
+        verify=False
+    )
+    if r.status_code == 200:
```

Comment:
OPTIONAL: ``if r:`` is enough (the response evaluates to True when the request was successful). I would also rename ``r`` to ``response``.